code/modules/hydroponics/grown/towercap.dm

```diff
@@ -179,6 +182,48 @@
 	return ..()
 
 /obj/structure/bonfire/attackby(obj/item/W, mob/user, params)
+	//SMOKE SIGNALS. Burn cloth to send a radio message to everyone who knows tribal.
+	if(istype(W, /obj/item/stack/sheet/cloth) && !grill && burning)
+		var/obj/item/stack/sheet/cloth/C = W
+		if(!user.has_language(/datum/language/tribal))
+			return ..()
+		var/outdoors = FALSE
+		for(var/area_type in OUTDOOR_AREAS)
+			if(istype(get_area(src), area_type))
+				outdoors = TRUE
+				break
+		if(!outdoors)
+			to_chat(user, span_warning("You must be outside to send a smoke signal."))
+			return
+		var/signalmessage = stripped_input(user, "What would you like to send via smoke signal?", "Smoke Signal")
+		if(!signalmessage)
+			return
+		var/send_time = length(signalmessage) * 2 //Deciseconds. A 10 character message would take 20 deciseconds, or two seconds, to send.
+		if(!user.canUseTopic(src, BE_CLOSE))
+			to_chat(user, span_warning("You have to stand still to send a smoke signal."))
+			return
+		if(do_after(user, send_time, target = src))
+			if(!C.use(1))
+				to_chat(user, span_warning("You don't have enough cloth to send a smoke signal."))
+				return
+			smoke_signal(user, signalmessage, src)
+			user.visible_message("[user] burns \the [W] in short, controlled bursts.")
+		else
+			to_chat(user, span_warning("You have to stand still to send a smoke signal."))
+		return
+	//SANDSTONE GUARDRAIL. Use sandstone to make the bonfire dense to avoid accidentally walking onto the fire.
+	if(istype(W, /obj/item/stack/sheet/mineral/sandstone) && !stones)
+		var/obj/item/stack/sheet/mineral/sandstone/S = W
```

Comment:
I'd prefer a more descriptive var than S. Trying to get away from one-letter vars.

That said, thank you thank you thank you for this change~

Reply:
The main reason it's a one-letter var is just because this is a typecast used to make sure the `use()` proc properly uses one sheet of sandstone. That being said, I can expand it!

Reply:
I know, I just really dont like one-letter vars. Not that big a problem here tbh.